### ISSUE TITLE
Driver deadlock

### ISSUE THREAD

**Issue by michael-fadely** [ViGEm/ViGEmBus#22]
**Describe the bug**
With a virtual Xbox 360 controller enabled, programs that poll controller input will be unable to launch. They deadlock attempting to open a handle to the device. This happens with feeder applications as well.

It also prevents any real devices from connecting to or disconnecting from my system.

**To Reproduce**
I've been unable to consistently reproduce it. It happens fairly frequently, but with no specific steps that I could identify. If applications begin failing to launch with a feeder and Xbox 360 endpoint, however:
1. Launch an application that polls for controller input (a game, Uplay, Steam, Battle.net).
2. The application will stall or fail to launch.
3. Attempt to disable the device with Device Manager/etc, and it will also stall.
4. When the feeder is killed, the Xbox 360 endpoint device will enter an error state, displayed in Device Manager.

**Expected behavior**
Some kind of error handling or error logging. ViGEmClient has error logging, but it doesn't appear to encounter any errors when this deadlock occurs.

**Screenshots**
![image](https://user-images.githubusercontent.com/920569/62824650-7d004380-bb5e-11e9-9231-dc600d80f83c.png)

I attempted to disable the device, Device Manager stalled, and both the Game Controllers dialog and Battle.net immediately launched. Opening another instance of Device Manager reveals that the device is still enabled:

![image](https://user-images.githubusercontent.com/920569/62824701-29dac080-bb5f-11e9-8333-31f3779a669e.png)

**System details (please complete the following information):**
 - OS: Windows 10 Pro 1903
 - Feeder software: [ds4wizard](https://github.com/michael-fadely/ds4wizard/)
 - Driver Version: 1.14.3.0

**Additional context**
It very well could be something I'm doing with my feeder, but I've been unable to figure it out.

**Comment by nefarius** [comment]
Thanks for the report but could you please monitor the situation with the latest driver version as well? What you run is very outdated.

**Comment by michael-fadely** [comment]
...Wow, that IS really old! I missed the latest stable release somehow. Will do!

**Comment by michael-fadely** [comment]
Same issue with latest stable. I will do some experimenting to see if I can narrow down the issue.

**Comment by MtDewFella** [comment]
I have been getting the same problem, wasn't sure if it was steam, glosc or vigem, I assume it's vigem since trying to disable it causes driver manager to freeze up, only fix I have found is to reboot the pc, oddly it never happens during play only after (not all the time only some times)

**Comment by nefarius** [comment]
Dammit 😅

**Comment by MtDewFella** [comment]
I completely uninstalled vigem and glosc, and reinstalled vigem first then glosc and so far I have had no issues, the last time I installed I just installed glosc like normal, which installed vigem (or was supposed to) but when I first used glosc that time I got the couldn't initialize vigem error, I then installed vigem without uninstalling anything, perhaps that caused the issue, I will keep and eye on it

**Comment by nefarius** [comment]
Any news @michael-fadely ?

**Comment by michael-fadely** [comment]
Still exhibiting the exact same behavior, unfortunately. Just having the virtual Xbox 360 device present when attempting to launch an application that uses XInput will *sometimes* cause this deadlock and require a hard reboot to allow any other physical devices to connect to, or disconnect from, the system.

**Comment by mosihe** [comment]
> Still exhibiting the exact same behavior, unfortunately. Just having the virtual Xbox 360 device present when attempting to launch an application that uses XInput will _sometimes_ cause this deadlock and require a hard reboot to allow any other physical devices to connect to, or disconnect from, the system.

I have the similar issue.
I want to know if your game program exits normally.
And How often does this issue occur.

**Comment by michael-fadely** [comment]
Could this be due to my program's multi-threading? I am using [ViGEmClient](https://github.com/ViGEm/ViGEmClient). I call `vigem_alloc` and `vigem_connect` (as well as free and disconnect) on the main thread, but `vigem_target_add`, `vigem_target_x360_register_notification`, etc. are called on a secondary thread.

**Comment by mosihe** [comment]
Hello, I also have the problem with a single thread.I now think that it should be a better way to reproduce the problem scene, although this is difficult.
In order to narrow the scope of the system environment, I want to know how often you will get an exception and whether the process will be forced kill.


mo.he@vrviu.com
 
From: Michael Fadely
Date: 2020-01-30 06:04
To: ViGEm/ViGEmBus
CC: mo.he-vrviu; Comment
Subject: Re: [ViGEm/ViGEmBus] Driver deadlock (#22)
Could this be due to my program's multi-threading? I am using ViGEmClient. I call vigem_alloc and vigem_connect (as well as free and disconnect) on the main thread, but vigem_target_add, vigem_target_x360_register_notification, etc. are called on a secondary thread.

**Comment by michael-fadely** [comment]
In my program, everything appears to continue working. I simply unplug the DualShock 4 and it calls `vigem_remove_target` seemingly without error. I will implement some more robust error checking in this instance to be certain.

**Comment by Corrodias** [comment]
This seems very similar to a problem I'm experiencing. i'm using DS4Windows, on Windows 10 1909. After about 3 days of uptime, Monster Hunter World will stall on its black startup screen and not proceed to the dev logo, but Special K's overlay is drawn, and Chrome can still start a new, GPU-accelerated video, so it's not a video driver failure. Attempting to "stop" and "start" DS4Windows may appear to work but results in a CPU-spinning thread. Attempting to reboot Windows causes it to hang on "Restarting" for several minutes, then BSOD.

I have **not** yet narrowed this down to being caused by ViGEmBus. It does seem like a likely culprit, especially after reading this thread. I'll try uninstalling it after the next time this happens. I will also attempt to get an xperf trace on the shutdown process that includes stacks when it's stuck.

**Comment by Corrodias** [comment]
In fact, if it's going to do it again, today's the day. I have about 7 or 8 hours before I'm going to commence playing games, at which time I'll be doing the trace and so on if it did deadlock. If you have any advice or suggestions on helping diagnose this, providing them now would be timely.

**Comment by nefarius** [comment]
Make sure kernel memory dumps are enabled and keep the dump file somewhere. It shouldn't be shared publicly but is the best start to get the culprit.

**Comment by Corrodias** [comment]
Roger. Seems I'll have to reboot to enable that, so I'll get it ready for next time. I wouldn't get my hopes up, though; it didn't record the bugcheck event to the log the last time, despite the BSOD. (And it rebooted to BIOS setup instead of Windows for some reason; I was able to just exit with no changes and go right back to Windows.) I could only guess as to why. It wrote one of those "Dump file creation failed" events after booting up.

Since the last time it happened, I removed an old Logitech driver I had and replaced it with a newer one. If that was the actual culprit, I won't be seeing this situation again.

**Comment by Corrodias** [comment]
It happened again today, so I took some traces. I'll link them to you in an email.

I launched MHW, and it froze on its black startup screen. I ran a trace and produced "out.etl". Upon unplugging the controller, DS4Windows reported losing connection to it, and MHW suddenly continued as normal. I was also able to exit and launch it again normally. Plugging the controller back in, DS4Windows did not log any new messages, such as connecting to it.

Attempting to restart got stuck, as before. I took a trace during that restart, and I hit reset after it was clear that it wasn't going anywhere, rather than let it play out for 10 minutes. Unfortunately, xbootmgr was not able to assemble a finalized trace file once Windows started up; it claimed it couldn't load something from the registry. Therefore, you have "kernel.etl", "shutdown_BASE+DIAG+LATENCY+PROFILE+INTERRUPT+DPC_1_um_premerge.etl", and "shutdown_BASE+DIAG+LATENCY+PROFILE+INTERRUPT+DPC_1_km_premerge.etl".

There's also the less-inclusive "shutdown_BASE+DIAG+LATENCY_1.etl" that is from a previous attempt at a restart trace during this stuck condition.

If it happens again in a few days, I now have kernel memory dumping enabled, so I'll force a bugcheck in the hopes that'll actually dump it, following Microsoft's instructions for doing so.

**Comment by Corrodias** [comment]
The deadlock has not yet triggered again, but I'm still being vigilant and and ready to catch a memory dump when it does.

**Comment by Corrodias** [comment]
Damn, failure. It finally happened again, and I initiated a crash, but Windows once again failed to write a dump file due to an unspecified error. It looks like the only other option for getting a kernel memory dump is enabling kernel debugging mode and dumping it with windbg. I'll try to get that set up for next time.

**Comment by nefarius** [comment]
@Corrodias shame, but was it "random" or while a game accessed the controller? Because I was thinking about writing a small "fuzzing" application which may get us to the goal faster. Like rapid (dis-)connects, calling XInput APIs automated many many times etc. Might be with the try other than waiting and hoping 😅

**Comment by Corrodias** [comment]
I am hesitant to enable kernel debugging for several days at a time because I'm wary of a performance impact, and Monster Hunter: World is already a very demanding game. Having a way of reproducing it that doesn't require waiting upward of 3 days would be excellent. I'd be eager to try such an application and wouldn't have a reason to avoid the debugging mode then.

I only notice it happening right as the game's initializing, while the window is black and Special K has drawn its banner, but before it displays the splash screens, most likely during xinput initialization. That said, I can't guarantee that the truth isn't that the state is being triggered before that and the game is merely _encountering_ it. I'm using DS4Windows with "exclusive mode" enabled, so the game cannot see the original DS4.

This may be important: my DS4 is actually connected through a third-party adapter called the Titan Two. That's connected to the PC by USB at all times, and I merely swap the controller between bluetooth and USB. It's supposed to perfectly emulate a DualShock 4 and by all appearances is successful in doing so. However, if it deviates in some small way, perhaps that could be triggering this issue? I could try connecting the controller directly by USB for a couple of weeks to see if that prevents the problem. However, if the Titan Two is misbehaving in some small way, I think it would benefit both its devs and you to understand how and why it's causing a problem, so one or both ends can mitigate it.

Thinking back, unfortunately, I see no pattern between things I do before a gaming session that works and things I do before one that doesn't. It's all the same stuff. I connect and disconnect my USB headset on occasion. I swap the controller between bluetooth and USB so it can charge.

**Comment by Corrodias** [comment]
I have the XinputTest program from x360ce, so I can try opening and closing that 30 times under various conditions. I did mange to trigger it now but not on a second attempt. Maybe I have to have Steam open. Maybe I have to swap to bluetooth once first. I have to wait until after my pirate anime stream with friends before I can continue testing, but I'll see if I can get this thing to trigger manually.

**Comment by Corrodias** [comment]
Things that don't trigger it:
Running XinputTest 60 times in a row, while Steam's running.
Running MHW 14 times.
Switching profiles in DS4Windows 20 times.
Running MHW while there are 15 instances of XinputTest open.

There really seems to be a time component to this, somehow. It just won't do it for me on demand.

**Comment by Corrodias** [comment]
I gave in and enabled kernel debugging. Surprisingly, it seems to have no appreciable effect on performance, at least when the debugger isn't attached. Nice!

My plan is to run livekd64.exe and run a full dump from kernel mode in kd (with /f), while the system's running, the next time it happens. That'll be 32 GB, since that's how much RAM I have, and apparently when you tell it you want a "full" dump, you mean **full**.

Would you like me to use any other options instead?

**Comment by Corrodias** [comment]
Starting to think that the game running in DX12 mode just doesn't trigger it, so I'm going to switch back to DX11 mode. Shrug emoji.

**Comment by michael-fadely** [comment]
A combination that tends to have this issue a lot for me is uplay + Rainbow Six: Siege, if you've got that handy.

**Comment by Corrodias** [comment]
@michael-fadely I don't, unfortunately. If you can reproduce it as well, then you could submit a memory dump.

I'm still waiting for it to happen, and I fear it won't with DX12 mode, but for unrelated reasons, I can't play in DX11 mode right now.

There are 2 ways:
1) The simple way, if it works. Configure Windows to do a kernel memory dump when it crashes, then manually crash it when the condition occurs. Instructions here: https://github.com/Ryochan7/DS4Windows/issues/1049#issuecomment-590027280 - The downside is that on some systems (including mine), Windows never writes memory dumps. Seems to be a common thing when you're booting from an SSD.
2) Install the Windows debugging kit, enable local kernel debugging with bcdedit, reboot, wait for the condition to occur, and use livekd to do a full memory dump, then send that to nefarius. This is a bit more work, but I have confirmed that there seems to be zero performance impact from enabling debugging. The main downsides here are uploading a gigantic dump file and giving nefarius access to anything secret you have in memory. I trust him with it.

**Comment by AugmentedAntics** [comment]
Well, after several weeks of thinking I found a way around it, the deadlock started happening again. And this is the fastest record it's had, less than 1 day after the last reboot and it happens after a few hours in Dolphin.  I do have a memory dump after waiting for this to happen again, so if it helps, I can send it.

**Comment by Corrodias** [comment]
Please do send it to nefarius. I assume the email address in their profile is accurate.

@nefarius 
I, too, have finally reproduced the issue, and will be sending an S3 download link. While MHW never triggered it again, restarting Steam did.

**Comment by AugmentedAntics** [comment]
Alright, I sent it to him.

**Comment by nefarius** [comment]
Received and will have a look, thanks for your involvement!

**Comment by AugmentedAntics** [comment]
Did the file sharing work?

**Comment by nefarius** [comment]
Yes, downloading right now!

**Comment by Corrodias** [comment]
Just to clarify, mine is a live dump using livekd that was taken while Steam was in the state of trying to start but being stuck. I had not yet unplugged the controller.

**Comment by krzesu0** [comment]
I've been having the same issue lately. Some days it would happen 3 to 4 times per day. I've changed to which port I connect the Steam Controller from USB 3.0 Hub to direct 2.0 port. No issues from then.
Using 1.16.116 with GloSC. _Maybe this will help somehow._

**Comment by nefarius** [comment]
I'm back on the hunt 🤠

**Comment by mosihe** [comment]
> 🤠

 🤠

**Comment by nefarius** [comment]
I'll need test personnel once I baked something 🙃

**Comment by mosihe** [comment]
> I'll need test personnel once I baked something 🙃

I am not very clear what you mean,
I can reproduce the deadlock every day,
And happy to help.

**Comment by nefarius** [comment]
@mosihe I'd need people with the issue (like you) to put their system into testmode and run a new driver version for testing once I have binaries is what I mean 😅

**Comment by mosihe** [comment]
> @mosihe I'd need people with the issue (like you) to put their system into testmode and run a new driver version for testing once I have binaries is what I mean 😅

ok, so where to download the new driver, or you mean compile the master driver version.

**Comment by nefarius** [comment]
Hold your horses dude, I haven't even started yet, I'd announce it if there was anything to download and where from 😉

**Comment by nefarius** [comment]
Ahoy! [Beta build with attempted fix is online here](https://buildbot.vigem.org/builds/ViGEmBus/feature/core/cpp-ify/1.17.241.0/bin/), fee free to try, just update it over the existing `Virtual Gamepad Emulation Bus` device in device manager. **Testsigned** so **testmode is mandatory!**

Cheers

**Comment by AugmentedAntics** [comment]
So, what can be done if Windows nags that the "best driver is already installed"?

**Comment by nefarius** [comment]
You force it. 

![image](https://user-images.githubusercontent.com/286631/81845469-e3344f80-9550-11ea-806f-a9eec807c023.png)

Navigate to the folder containing all the downloaded files, choose the single arch INF file and select that, acknowledge the warning dialog and you're good to go.

**Comment by AugmentedAntics** [comment]
I have successfully installed the test driver, but unfortunately, the deadlocking is still an issue. As I'm typing this, having DS4Windows open while it is occurring stalls any applications looking for the controller, and I already know restarting my PC will also stall. I spent a few hours in Citra prior to it.

**Comment by nefarius** [comment]
Are you a 100% sure that the beta driver is loaded, can you confirm with screenshots of device manager version please.

**Comment by AugmentedAntics** [comment]
![Annotation 2020-05-14 143817](https://user-images.githubusercontent.com/54159324/81988653-acae1180-95f0-11ea-92e9-4a0b43e47c8d.png)

This is what it looks like for me.

**Comment by nefarius** [comment]
*Sad noises* your view is correct. It happens while emulating x360 right?

**Comment by AugmentedAntics** [comment]
Yes, that's right.

**Comment by nefarius** [comment]
Meh, so the search continues.

**Comment by nefarius** [comment]
@AugmentedAntics are you on the Discord btw.? Might need to change my investigation strategy and I need affected users since I can't replicate.

**Comment by AugmentedAntics** [comment]
I am. I go by the same name, except there's a space between Augmented and Antics.

**Comment by Corrodias** [comment]
I'm willing, but since the problem has become much rarer than it used to be for me (I've gone weeks of uptime without seeing it), and I have found no method of reproducing it, I can't guarantee that my participation would be useful. Corrodias#2162

**Comment by nefarius** [comment]
[Here's the latest build with quite a few promising improvements.](https://buildbot.vigem.org/builds/ViGEmBus/feature/core/cpp-ify/1.17.258.0/disk1/)

**Comment by nefarius** [comment]
Nope, still happens. Slowly but steadily running out of ideas.

**Comment by gjmv** [comment]
Hi, i have a similar issue. 
I'm using GloSc and frequenly when i close the overlay, my device manager is frozed up, and the Steam Controller don't work anymore.
I have to reset the PC, it stucks in "rebooting", then i manually push reset button. (plus others problems like said in the thread)
Yesterday i removed glosc and vigembus driver, then install latest vigembus driver first, then install glosc, and today the issue occurs 2 times. (before this happened every 2 o 3 days, casuality?)
My system never crash (never BSOD) but needs a fisical reboot to get working again.
I forced a BSOD and send dump file to @nefarius 

This is an image of my device manager

![device manager](https://user-images.githubusercontent.com/20653690/85500923-c6ad1d80-b5ba-11ea-801c-b014178000c5.png)

Thanks!

**Comment by nefarius** [comment]
Thanks @gjmv I could in the meantime sort of replicate it and I think the culprit is the ironically the mechanism to keep the plug-in request on hold until the device is ready. Although I haven't found a clear indicator of an issue within the logic itself, I have the suspicion that it somehow causes a race condition in conjunction with MS `xusb.sys` and stalls unrecoverable. I may need to redesign the entire plugin API but other than backwards compatibility that's necessary anyway, I don't like how it was designed in the first place.

**Comment by michael-fadely** [comment]
How do you currently replicate it?

I've run into this issue pretty persistently with Dark Souls 3 and would be willing to do some debugging of my own. Tangentially because I'm experiencing input lag in this game using my feeder, but that might be my fault. 😄

**Comment by nefarius** [comment]
Instead of [doing this](https://github.com/ViGEm/ViGEmBus/blob/9cbd4b65c2f9e017cd3918f3191341e80ee1df30/sys/busenum.cpp#L222-L224), just return `STATUS_SUCCESS` basically.

**Comment by michael-fadely** [comment]
Nice. I'll have a look if I can find the time.

**Comment by KerrmexRPG** [comment]
I'm experiencing this problem too. Windows 10 2004 fresh install with a DS3 controller here...

**Comment by Corrodias** [comment]
@michael-fadely Take care not to overestimate input latency in Dark Souls 3. There's something upward of a 250 ms delay on rolling, just the way the game is made. It's frustrating at times, as a player, as you often have to time rolls before enemies swing; it's too late to roll once they start swinging. I'm sure you can compare latency using other games or other input methods, and I have no doubt you're capable; just making sure.

**Comment by michael-fadely** [comment]
@Corrodias lol yep, I know. Thanks for the heads up though!

**Comment by nefarius** [comment]
Goo day, people who've been bugged by this, err, bug 😁 May you please check out the new release 😘

**Comment by michael-fadely** [comment]
Same issue with my feeder application, but I was using ViGEmClient commit `465736429b8fe2b9d236b01ef0404f9bceb31106`. I will test with the latest commit on branch `v1.17-experimental` to be certain. Driver version is reporting 1.17.333.0 in Device Manager.

**Comment by gjmv** [comment]
I've been playing for 4 days with no problems or deadlocks (GloSC).
I'm using Release version 1.17.333.

**Comment by AugmentedAntics** [comment]
Few days have passed for me, and no deadlocks yet.

**Comment by Corrodias** [comment]
I am not presently playing a game with a gamepad, but I'll let it run, and once I get back to it, I'll chime in.

**Comment by AugmentedAntics** [comment]
Well, unfortunately, it's happened again anyway. I'll have to fall back to just reinstalling the old 2009 360 driver, since that seemed to ward off the deadlocks completely for some reason.

**Comment by KerrmexRPG** [comment]
> Well, unfortunately, it's happened again anyway. I'll have to fall back to just reinstalling the old 2009 360 driver, since that seemed to ward off the deadlocks completely for some reason.

How do you install the old driver? Is it possible in Windows 10? Where can i download that 2009 driver? 
Thanks

**Comment by nefarius** [comment]
Interesting hint, I'll see if I can find some wisdom in the different versions after the holidays.

**Comment by nefarius** [comment]
> > Well, unfortunately, it's happened again anyway. I'll have to fall back to just reinstalling the old 2009 360 driver, since that seemed to ward off the deadlocks completely for some reason.
> 
> How do you install the old driver? Is it possible in Windows 10? Where can i download that 2009 driver?
> Thanks

[Should be available here](https://www.catalog.update.microsoft.com/Search.aspx?q=Xbox%20360).

**Comment by rolibra** [comment]
Hi,

My pc is also getting stuck on reboot.
I'm using steamlink + playnite +gloSC to play pc games on the tv.

Running driver version 1.17.333.

**Comment by nefarius** [comment]
Send me your PC, I wanna finally find out what's wrong here 😅

**Comment by KerrmexRPG** [comment]
> Send me your PC, I wanna finally find out what's wrong here 😅

After a couple of days with the 2009 driver installed I can confirm that the deadlock issue is gone. Everything is working perfectly fine now.

**Comment by rolibra** [comment]
> > Send me your PC, I wanna finally find out what's wrong here 😅
> 
> After a couple of days with the 2009 driver installed I can confirm that the deadlock issue is gone. Everything is working perfectly fine now.

Any specific version of the driver?



> Send me your PC, I wanna finally find out what's wrong here 😅

Haha.

Sorry, please let me know of any info/logs i can send as evidence.

**Comment by KerrmexRPG** [comment]
> > > Send me your PC, I wanna finally find out what's wrong here 😅
> > 
> > 
> > After a couple of days with the 2009 driver installed I can confirm that the deadlock issue is gone. Everything is working perfectly fine now.
> 
> Any specific version of the driver?
>
![image](https://i.imgur.com/0GM2Blm.jpg)

**Comment by Corrodias** [comment]
So far, my PC hasn't misbehaved, with the default win10 driver for xbox gamepads, version 10.0.19041.1 from 2019-12-06. But I haven't been _using_ the gamepad, and I'm not sure whether that makes a difference. A couple of days may not be enough; I've had it work for a week before failing, in the past. That said, I remember people saying back then that using the older xbox driver helped then, too.

**Comment by rolibra** [comment]
Ok, i had the issue again, and just noticed conflicting devices on the device manager:

![image](https://user-images.githubusercontent.com/76675035/103971873-6e9aca80-514a-11eb-9bab-6de5319e5e85.png)

**Comment by nefarius** [comment]
That's not conflicting, that means the device is in error state, double-click to see the error message and code it reports.

**Comment by KerrmexRPG** [comment]
No deadlocks for me since I installed the 2009 driver, I'm just saying... I've been using my controller often in these days.

**Comment by rolibra** [comment]
I think i've identified a pattern on the issue for me.

If I play using 1 controller only, no issues. But If i play a co-op game using 2 controllers (Streets of rafe 4, tetris), then i get the issue when shutting down or rebooting.

**Comment by Corrodias** [comment]
I played Horizon Zero Dawn on and off this week without issue. Finally rebooted today for updates, which went fine.

For those of you still experiencing the problem, what driver version for the Xbox 360 Controller do you have? Looking at that properties dialog, the date and version number are the useful information. I already said what mine is, and I don't seem to be experiencing the issue any more.

**Comment by Corrodias** [comment]
Finally experienced the deadlock again, so this combination with the newest 360 driver is unfortunately still a problem.

**Comment by gjmv** [comment]
I have installed the new version since December, and it only happened to me 2 times the first days. During January and February I used it a lot, and I didn't have any deadlocks.
Thanks for the update!
Regards

**Comment by nefarius** [comment]
Even once is too much though, thanks for the feedback nonetheless!

**Comment by Corrodias** [comment]
Pity, I left it running overnight ONE DAY and it happened again. 😆 

Is there any thing further I can do to help you diagnose this, other than shipping you my PC? I provided some (unfortunately massive) memory dumps, but evidently those weren't enough on their own.

**Comment by nefarius** [comment]
> Pity, I left it running overnight ONE DAY and it happened again. 😆
> 
> Is there any thing further I can do to help you diagnose this, other than shipping you my PC? I provided some (unfortunately massive) memory dumps, but evidently those weren't enough on their own.

I'll just scratch everything and start anew 👻

**Comment by valentyn-l** [comment]
I am STILL getting a driver deadlock issue with the 2021 Xbox 360 controller driver. Not an issue with the 2009 driver. This is a problem though because the 2009 driver prevents me from activating Core Isolation in the Device Security settings. Tested this on both Windows 10 and 11 leaked build.

**Comment by nefarius** [comment]
I've pretty much given up on this to be honest. There's simply no resources I or any potential contributor could come up with to tackle this mystery and I'm tired of it and will move on. Sorry.

**Comment by andrewklamut** [comment]
Hi, I'm facing a 100% reproducible issue which might be relevant? The issue is that when I start to use my Xbox 360 controller over Parsec, the host system hangs and must be forced shut down.

Steps:
1) Using Windows Hyper V, create a Windows 10 VM
2) Install Parsec on VM guest and VM host
3) Plug Xbox 360 controller into VM host
3) Start the Windows 10 VM guest using Hyper-V -> Start (don't do Hyper-V connect) and connect via Parsec
4) Press any button on the Xbox 360 controller

5) The VM guest hangs and must be force shut down. If I open Device Manager on the VM guest before step 4, I can see the Xbox Peripheral device is added as step 4 happens, and the Device Manger refreshes very slowly (I see the tree collapse, and then expand).

It kind of seems like a deadlock and I came across this thread and noticed you were stuck, maybe same root cause?

**Comment by wingfixer** [comment]
Found this topic by accident, realized it was the same issue I had with (as I thought) DS4Windows forever ago, ended up giving up on it and just setting a hotkey to disable the whole thing when not needed.
Appears to be completely random, could run for hours or days before it happens, so people that don't leave their PC running for long may be susceptible but will never notice it.

**Comment by nefarius** [comment]
It will be solved by the rewrite I am working on in the dev dungeon ⚒️

**Comment by nefarius** [comment]
Time to close this. Can't fix something I can not replicate. Sorry.

**Comment by gjmv** [comment]
I have the driver installed again in windows 11, in PC and laptop, and never again get bluescreen.

**Comment by nefarius** [comment]
This issue isn't about a bluescreen.

**Comment by gjmv** [comment]
> This issue isn't about a bluescreen.

Sorry, deadlock

> I have installed the new version since December, and it only happened to me 2 times the first days. During January and February I used it a lot, and I didn't have any deadlocks. Thanks for the update! Regards
